### ISSUE TITLE
ObjectListOption.prefix() returns nothing

### ISSUE THREAD

**Issue by dri94** [storj/uplink-android#3]
If I upload an object with the options `path = /test/filename.jpeg` and path `/test/again/filename.jpg` then they do not seem to be correctly returned. When i call `bucket.listObjects()`, I get back a `ObjectInfo` that has `path = /`. If I then call `bucket.listObject(ObjestListOption.prefix("/"))`, I get back no results. Is there a bug in the code or am I doing something wrong?

I imagine it all to be like the following analogy. A bucket is a hard drive in a computer. You can have multiple hard drives. A path is a specified location in that hard drive, and the full path will always be unique among files.

**Comment by kaloyan-raev** [comment]
We haven't updated the Android library for more than 3 months. We plan update for next week. Hopefully, this would be something we fixed lately because I don't recall such an issue in our Go library.

**Comment by dri94** [comment]
Okay so with the same code and encryption key, It loads the initial list like before, but now when trying to load objects with the prefix "/" i get an exception `io.storj.StorjException: go.Universe$proxyerror: kvmetainfo: encryption error: decryption failed, check encryption key: cipher: message authentication failed`.. I am using the same encyrption key, why would it throw this error? @kaloyan-raev

**Comment by kaloyan-raev** [comment]
Yes, this happens with the latest 0.12.0 release. Updating the Android libraries to latest changes did not resolve the issue, it just changed the erroneous behavior. We have issues with some corners cases with leading and trailing slashes. We are working on finding a good solution. I will update here when we have a working solution.

BTW, I am curious to hear about your use case that requires a leading slash for the files in the bucket.

**Comment by dri94** [comment]
It doesnt require it, are you saying removing that slash would make this work? Would I be able to have slashes in between characters?
